specification/cognitiveservices/data-plane/Language/preview/2022-02-01-preview/textanalytics.json

```diff
@@ -0,0 +1,3090 @@
+      }
+    },
+    "MultiLanguageAnalysisInput": {
+      "properties": {
+        "documents": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/MultiLanguageInput"
+          }
+        }
+      }
+    },
+    "LanguageDetectionAnalysisInput": {
+      "properties": {
+        "documents": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/LanguageInput"
+          }
+        }
+      }
+    },
+    "AnalyzeTextTask": {
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeTextTaskKind"
+        }
+      }
+    },
+    "AnalyzeTextTaskResult": {
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeTextTaskResultsKind"
+        }
+      }
+    },
+    "AnalyzeTextEntityLinkingInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/EntityLinkingTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "EntityLinking"
+    },
+    "AnalyzeTextEntityRecognitionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/EntitiesTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "EntityRecognition"
+    },
+    "AnalyzeTextKeyPhraseExtractionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/PreBuiltTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "KeyPhraseExtraction"
+    },
+    "AnalyzeTextPiiEntitiesRecognitionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/PiiTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "PiiEntityRecognition"
+    },
+    "AnalyzeTextLanguageDetectionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/LanguageDetectionAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/LanguageDetectionTaskParameters"
+        }
```

Comment:
For the input for language detection tasks, should a user specify the country_hint in LanguageDetectionAnalysisInput or the region_hint in LanguageDetectionTaskParameters? Are these the same?

Reply:
To be honest, I don't remember why I added `regionHint` as a task parameter.  We can use `countryHint` in the language batch input as usual instead.  I am in the process of removing it and minorly restructuring the task parameters into separate objects for easier reuse.  I will push the change shortly.

Reply:
This is now updated.  I've removed `regionHint` entirely, and also restructured task parameters slightly to make them more reusable.  Let me know what you think.